File: jdroid-android-core/src/main/java/com/jdroid/android/firebase/analytics/AbstractFirebaseAnalyticsTracker.java
package com.jdroid.android.firebase.analytics;

import com.jdroid.java.analytics.AnalyticsTracker;

import java.util.concurrent.Executor;

import androidx.annotation.WorkerThread;

public class AbstractFirebaseAnalyticsTracker implements AnalyticsTracker {

	@Override
	public Boolean isEnabled() {
		return true;
	}

	@WorkerThread
	protected FirebaseAnalyticsHelper getFirebaseAnalyticsHelper() {
		return FirebaseAnalyticsFactory.getFirebaseAnalyticsHelper();
	}

	@Override
	public Executor getExecutor() {
		return getFirebaseAnalyticsHelper().getExecutor();
	}
}

File: jdroid-android-core/src/main/java/com/jdroid/android/firebase/analytics/AbstractFirebaseAnalyticsTracker.kt
package com.jdroid.android.firebase.analytics

import com.jdroid.android.koin.KoinHelper
import com.jdroid.java.analytics.AnalyticsTracker
import org.koin.core.get
import java.util.concurrent.Executor

open class AbstractFirebaseAnalyticsTracker : AnalyticsTracker {

    override fun isEnabled(): Boolean {
        return true
    }

    protected fun getFirebaseAnalyticsFacade(): FirebaseAnalyticsFacade {
        return KoinHelper.get()
    }

    override fun getExecutor(): Executor {
        return getFirebaseAnalyticsFacade().executor
    }
}
